source/common/network/utility.cc

```diff
@@ -574,30 +631,32 @@ Api::IoCallUint64Result Utility::readFromSocket(IoHandle& handle,
       ASSERT(msg_len <= slice->len_);
       ENVOY_LOG_MISC(debug, "Receive a packet with {} bytes from {}", msg_len,
                      output.msg_[i].peer_address_->asString());
-      passPayloadToProcessor(msg_len, *slice, std::move(buffers[i]), output.msg_[i].peer_address_,
+
+      // Adjust used memory length and commit slice to buffer
+      slice->len_ = std::min(slice->len_, static_cast<size_t>(msg_len));
+      buffers[i]->commit(slice, 1);
```

Comment:
Not sure I follow why this is now needed.

Reply:
On platforms that do not support udp_gro, we will still be utilizing the recvmmsg optimization (i.e. under the `handle.supportsMmsg()`). This uses the "recvmmsg" syscall instead of "recvmsg" syscall.

Since, the `receiveMessage(...)` helper function only provides the implementation of a single recvmsg syscall (and not recvmmsg) , therefore this part is kept separate and does not use the helper function.